Modules/Registration/Metricsv4/test/itkEuclideanDistancePointSetMetricTest3.cxx

```diff
@@ -0,0 +1,220 @@
+
+  using PointsContainerType = itk::VectorContainer<long unsigned int, PointType>;
+  using PointsLocatorType = itk::PointsLocator<PointsContainerType>;
+  auto pointsLocator = PointsLocatorType::New();
+
+  auto fixedPoints = PointSetType::New();
+  fixedPoints->Initialize();
+
+  auto movingPoints = PointSetType::New();
+  movingPoints->Initialize();
+
+  // Threshold used to test the distance threshold functionality
+  double distanceThreshold = 0.5;
+
+  // Create a few points and apply a small offset to make the moving points
+  auto      pointMax = static_cast<float>(1.0);
+  PointType fixedPoint;
+  fixedPoint.Fill(0.0);
+  fixedPoint[0] = 0.0;
+  fixedPoint[1] = 0.0;
+  fixedPoints->SetPoint(0, fixedPoint);
+  fixedPoint[0] = pointMax;
+  fixedPoint[1] = 0.0;
+  fixedPoints->SetPoint(1, fixedPoint);
+  fixedPoint[0] = 0.0;
+  fixedPoint[1] = pointMax;
+  fixedPoints->SetPoint(2, fixedPoint);
+  if (Dimension == 3)
+  {
+    fixedPoint[0] = 0.0;
+    fixedPoint[1] = 0.0;
+    fixedPoint[2] = pointMax;
+    fixedPoints->SetPoint(3, fixedPoint);
+  }
+  unsigned int numberOfPoints = fixedPoints->GetNumberOfPoints();
+
+  PointType movingPoint;
+  for (unsigned int n = 0; n < numberOfPoints; ++n)
+  {
+    fixedPoint = fixedPoints->GetPoint(n);
+    if (n == 0)
+    {
+      movingPoint[0] = fixedPoint[0] + 0.5;
+      movingPoint[1] = fixedPoint[1] + 0.75;
+    }
+    else if (n == 1)
+    {
+      movingPoint[0] = fixedPoint[0];
+      movingPoint[1] = fixedPoint[1] + 0.25;
+    }
+    else if (n == 2)
+    {
+      movingPoint[0] = fixedPoint[0] + 0.25;
+      movingPoint[1] = fixedPoint[1];
+    }
+    if (Dimension == 3)
+    {
+      movingPoint[2] = fixedPoint[2] + 0.75;
+    }
+    movingPoints->SetPoint(n, movingPoint);
+  }
+
+  pointsLocator->SetPoints(movingPoints->GetPoints());
+  pointsLocator->Initialize();
+
+  std::vector<double> distanceArray;
+  for (unsigned int n = 0; n < numberOfPoints; ++n)
+  {
+    auto tempFixedPoint = fixedPoints->GetPoint(n);
+    auto pointId = pointsLocator->FindClosestPoint(tempFixedPoint);
+    auto closestPoint = movingPoints->GetPoint(pointId);
+    distanceArray.push_back(closestPoint.EuclideanDistanceTo(tempFixedPoint));
+    std::cout << n << " " << tempFixedPoint << " , " << closestPoint << " "
+              << closestPoint.EuclideanDistanceTo(tempFixedPoint) << std::endl;
+  }
+
+  // Test with identity transform
+  std::cout << "Testing with Translation Transform." << std::endl;
+  using TranslationTransformType = itk::TranslationTransform<double, Dimension>;
+  auto translationTransform = TranslationTransformType::New();
+
+  // Instantiate the metric
+  using PointSetMetricType = itk::EuclideanDistancePointSetToPointSetMetricv4<PointSetType>;
+  auto metric = PointSetMetricType::New();
+  metric->SetFixedPointSet(fixedPoints);
+  metric->SetMovingPointSet(movingPoints);
+  metric->SetMovingTransform(translationTransform);
+  metric->SetDistanceThreshold(distanceThreshold);
```

Comment:
Please exercise the `GetDistanceThreshold` method here using the `ITK_TEST_SET_GET_VALUE` macro.